Added clique_number, independence_number

dstahlke opened the pull request:
These simple implementations use `maximal_cliques` as the computational engine.  Perhaps there is a more efficient way, but it would be great to at least have these implemented.

gdalle commented on src/community/cliques.jl:
````diff
@@ -137,3 +137,112 @@ function maximal_cliques end
+ [1, 3]
+```
+"""
+function maximal_independent_sets end
````
why is this in the cliques file?

gdalle commented on test/community/cliques.jl:
```diff
@@ -52,4 +52,7 @@
     end
+
+    @test clique_number(cycle_graph(11)) == 2
+    @test independence_number(cycle_graph(11)) == 5
```
could you also add tests for the max indep set, perhaps in another file?

gdalle commented on test/community/cliques.jl:
```diff
@@ -52,4 +52,7 @@
       @test test_cliques(h, Array[[7, 4, 5], [2, 6], [3, 5], [3, 6], [3, 1]])
     end
+
+    @test clique_number(cycle_graph(11)) == 2
```
might be better to include in `test_cliques`?

gdalle reviewed (changes requested):
Thanks for the PR, sorry for the late review

dstahlke commented:
I've pushed an update that hopefully addresses these concerns.  (I rebased to master - hopefully that doesn't make the diffs too messy.)

After making a proper unit test for independent sets, I found I had to convert AbstractGraph to SimpleGraph in order to call complement.  Performance shouldn't be an issue since this data copy is negligible in comparison to the cost of the maximal_cliques call.  But I don't know if there is some other reason this would be problematic.

    @traitfn function maximum_independent_set(
        g::AG::(!IsDirected)
    ) where {T,AG<:AbstractGraph{T}}
        # Convert to SimpleGraph first because `complement` doesn't accept AbstractGraph.
        return maximum_clique(complement(SimpleGraph(g)))

gdalle commented:
I would add a warning to the docstring pointing to the conversion, but otherwise I'd say it's okay

Krastanov commented:
This looks good, and thankfully it has not bitrotten in the last two years. Thank you for the contribution!
